hudi-utilities/src/test/java/org/apache/hudi/utilities/TestHoodieSnapshotExporter.java

```diff
@@ -0,0 +1,227 @@
+    commonOpts.put(DataSourceWriteOptions.RECORDKEY_FIELD_OPT_KEY(), "_row_key");
+    commonOpts.put(DataSourceWriteOptions.PARTITIONPATH_FIELD_OPT_KEY(), "partition");
+    commonOpts.put(DataSourceWriteOptions.PRECOMBINE_FIELD_OPT_KEY(), "timestamp");
+    commonOpts.put(HoodieWriteConfig.TABLE_NAME, "hoodie_test");
+
+
+    cfg = new HoodieSnapshotExporter.Config();
+
+    cfg.sourceBasePath = basePath;
+    cfg.targetOutputPath = outputPath = basePath + "/target";
+    cfg.outputFormat = "json";
+    cfg.outputPartitionField = "partition";
+
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    if (spark != null) {
+      spark.stop();
+    }
+  }
+
+  @Test
+  public void testSnapshotExporter() throws IOException {
+    // Insert Operation
+    List<String> records = DataSourceTestUtils.convertToStringList(dataGen.generateInserts("000", 100));
+    Dataset<Row> inputDF = spark.read().json(new JavaSparkContext(spark.sparkContext()).parallelize(records, 2));
+    inputDF.write().format("hudi")
+        .options(commonOpts)
+        .option(DataSourceWriteOptions.OPERATION_OPT_KEY(), DataSourceWriteOptions.INSERT_OPERATION_OPT_VAL())
+        .mode(SaveMode.Overwrite)
+        .save(basePath);
+    long sourceCount = inputDF.count();
+
+    HoodieSnapshotExporter hoodieSnapshotExporter = new HoodieSnapshotExporter();
+    hoodieSnapshotExporter.export(spark, cfg);
+
+    long targetCount = spark.read().json(outputPath).count();
+
+    assertTrue(sourceCount == targetCount);
+
+    // Test snapshotPrefix
+    long filterCount = inputDF.where("partition == '2015/03/16'").count();
+    cfg.snapshotPrefix = "2015/03/16";
+    hoodieSnapshotExporter.export(spark, cfg);
+    long targetFilterCount = spark.read().json(outputPath).count();
+    assertTrue(filterCount == targetFilterCount);
+
+  }
+
+  // for testEmptySnapshotCopy
+  public void init() throws IOException {
+    TemporaryFolder folder = new TemporaryFolder();
+    folder.create();
+    rootPath = "file://" + folder.getRoot().getAbsolutePath();
+    basePath = rootPath + "/" + HoodieTestUtils.RAW_TRIPS_TEST_NAME;
+    outputPath = rootPath + "/output";
+
+    final Configuration hadoopConf = HoodieTestUtils.getDefaultHadoopConf();
+    fs = FSUtils.getFs(basePath, hadoopConf);
+    HoodieTestUtils.init(hadoopConf, basePath);
+  }
+
+  @Test
+  public void testEmptySnapshotCopy() throws IOException {
+    init();
+    // There is no real data (only .hoodie directory)
+    assertEquals(fs.listStatus(new Path(basePath)).length, 1);
+    assertFalse(fs.exists(new Path(outputPath)));
+
+    // Do the snapshot
+    HoodieSnapshotCopier copier = new HoodieSnapshotCopier();
+    copier.snapshot(jsc, basePath, outputPath, true);
+
+    // Nothing changed; we just bail out
+    assertEquals(fs.listStatus(new Path(basePath)).length, 1);
+    assertFalse(fs.exists(new Path(outputPath + "/_SUCCESS")));
+  }
+
+  // TODO - uncomment this after fixing test failures
+  // @Test
```

Comment:
Notice this is from original test for Snapshot copier, not sure about the historical issue for skipping it. I would slightly favor fixing it while you're at this stage so that we can claim the new "Exporter" is better tested than the old "Copier" :)

Reply:
I'm not quite sure what the comments here mean and what the implications might be. It may take officials to sort it out.